app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
+  var vat = require('lib/vat');
+
+  var Model = Backbone.Model.extend({
+    initialize: function (options) {
+      options = options || {};
+
+      this.sentryMetrics = options.sentryMetrics;
+      this.window = options.window || window;
+
+      this.fetch();
+    },
+
+    defaults: {
+      flowBegin: null,
+      flowId: null
+    },
+
+    fetch: function () {
+
+      var error;
```

Comment:
> More generally, I guess I don't really understand the use pattern of fetch on these models. Who would be expected to call fetch() and under what circumstances

Usually it's the entity that creates the model, directly after initialization. The reason for `fetch` is because it allows data to be populated from a remote resource. Using `fetch` to do initialization from the get go does smell of YAGNI in this case.

> The UniqueUserId model also has a synchronous fetch() method that seems to be called nowhere except its initializer, and I wonder if it would make more sense to just fold it into the initializer.

That's my fault on the UniqueUserId, I was probably having a "make everything consistent" moment. In both cases, initialization should probably just happen in the `initialize` function because no asynchronous requests are not made. The same goes for `user.js`.

Didn't mean to lead you astray.